link/perf_event.go

```diff
@@ -204,6 +217,25 @@ func getPMUEventType(pmu string) (uint64, error) {
 	return et, nil
 }
 
+// determineRetprobeBit reads a Performance Monitoring Unit's retprobe (numeric identifier)
+// from /sys/bus/event_source/devices/<pmu>/format/retprobe.
+func determineRetprobeBit(pmu string) (uint64, error) {
+	p := filepath.Join("/sys/bus/event_source/devices/", pmu, "/format/retprobe")
+
+	data, err := os.Open(p)
```

Comment:
I would use `ioutil.ReadFile()`, followed by `bytes.TrimSpace()`. (see `uint64FromFile()`)

Also, you forgot `data.Close()` here.